### PR TITLE
{montserrat,b612}: fix build

### PR DESCRIPTION
**Pull request by DeeUnderscore**
###### Description of changes
Fixes needed after #173430

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @drupol, we've talked about this in #nix:nixos.org recently

### CODE REVIEW

**Comment by drupol** [comment]
Ok let's be honest here... I definitely prefer using mkDerivation , it's clearer and you just copy what you need instead of copying everything and removing everything except the share directory.

**Comment by drupol** [comment]
There's also this font to fix : https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/data/fonts/b612/default.nix#L20

**Comment by drupol** [comment]
If you would ask me what would be my preference when it comes to such derivation, I would definitely choose [this way](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/hubot-sans/default.nix). Why? Because it's easier for a newcomer to understand what is going on, and there is no need to deal with extra commands.

I also think it's better because when using `fetchFromGitHub`, it uncompress the whole archive in `$out` (while usually when installing font, we only need a subset of those files; *.ttf only for example), while it's not necessary, and then afterwards, we need to clean out the extra files... which is IMHO cumbersome.

**Comment by DeeUnderscore** [comment]
The main reason to use this method is to avoid putting the files in the Nix store twice, since the `fetchFromGitHub` output, when used as `src`, still goes in the store, separate from the `mkDerivation` output. Normally, this makes sense, since the build step involves something like compilation of the source, but in this case `mkDerivation` is just discarding part of the `src` output, and the `src` is not used elsewhere anyway. 
 
Personally, I don't really have a strong opinion here, but I'm also not going to rewrite existing packages to be the other way, unless we actually have some established convention one way or the other (I'm unaware of such). 

I'll take a look at B612 later, unless someone else gets to it.

**Comment by drupol** [comment]
I understand and it make sense indeed... I'm confused now :)

Maybe we can apply a filter on `fetchFromGitHub`?

**Comment by DeeUnderscore** [comment]
> Maybe we can apply a filter on `fetchFromGitHub`?

We could do `sparseCheckout`, but that would force `fetchFromGitHub` to use `fetchgit` rather than `fetchzip`. The latter is usually simpler and probably involves less network traffic, and the former does involve a bunch of extra steps, too (like deleting the `.git` directory, which `fetchgit` does by default).

**Review by drupol — APPROVED** [review]

File: pkgs/data/fonts/b612/default.nix [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
+    shopt -s extglob dotglob
+    rm -rf $out/!(share)
+    shopt -u extglob dotglob
   '';
-  sha256 = "0r3lana1q9w3siv8czb3p9rrb5d9svp628yfbvvmnj7qvjrmfsiq";
+
+  hash = "sha256-aJ3XzWQauPsWwEDAHT2rD9a8RvLv1kqU3krFXprmypk=";
 
   meta = with lib; {
     homepage = "http://b612-font.com/";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/b612/default.nix:24]:
```suggestion
    homepage = "https://b612-font.com/";
```

**Pull request merged**